taskvine/src/manager/vine_schedule.c

```diff
@@ -211,9 +204,42 @@ int check_worker_against_task(struct vine_manager *q, struct vine_worker_info *w
 		}
 	}
 
+	if (t->needs_library) {
+		struct vine_task *library = vine_schedule_find_library(w, t->needs_library);
+		if (library) {
+			if (library->function_slots_inuse >= library->function_slots) {
+				return 0;
+			}
+		} else {
+			/* this is bad and we should feel bad. checking for matches should not modify the state of
```

Comment:
I agree this is a code smell and a hazard to the unwary, but it seems unavoidable with the current framework.